src/VehicleSelect/VehicleSelect.tsx

```diff
@@ -0,0 +1,221 @@
+import Autocomplete from "../Autocomplete";
+import { Box } from "@mui/material";
+import React from "react";
+import { VehicleSelectProps } from "./VehicleSelect.types";
+
+// component to select a vehicle
+function VehicleSelect({
+  allVehicles = [],
+  selectedVehicles = [],
+  flexDirection = "column",
+  flexWrap = "nowrap",
+  allGates = [],
+  onVehicleChange = () => {}
+}: VehicleSelectProps) {
+  // derive state for project
+  const selectedProjects = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.project))
+  ];
+  if (selectedProjects.length > 1)
+    throw new Error("Project selection is ambiguous");
+  const selectedProject = selectedProjects[0] ?? "";
+
+  // derive all projects
+  const allProjects = [
+    ...new Set(allVehicles.map(vehicle => vehicle.projectCode))
+  ].sort();
+
+  // derive state for model year
+  const selectedModelYears = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.modelYear))
+  ];
+  if (selectedModelYears.length > 1)
+    throw new Error("Multiple year is ambiguous");
```

Comment:
same comment here. 
Do we need this error checking because we never come across this situation right? As it is a single selection auto-complete there is no way we will have length > 1 Though we have duplicates in the object.

Reply:
This check will make the component a more robust